### PR TITLE
Neto Provider

### PR DESCRIPTION
**Pull request by berestniev**
Neto Provider

### CODE REVIEW

File: src/Neto/Provider.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
+    {
+        return array_merge(parent::getTokenFields($code), [
+            'grant_type' => 'authorization_code',
+        ]);
+    }
```
Comment (atymic) [inline src/Neto/Provider.php:81]: Need to add this so it loads the config correctly.

```
/**
     * {@inheritdoc}
     */
    public static function additionalConfigKeys()
    {
        return ['app_portal_domain'];
    }
```

File: src/Neto/Provider.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'NETO';
```
Comment (lucasmichot) [inline src/Neto/Provider.php:13]:
```suggestion
    public const IDENTIFIER = 'NETO';
```

File: src/Neto/Provider.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        return 'https://'.$this->getConfig('app_portal_domain').'/oauth/v2/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get('https://'.$this->getConfig('app_portal_domain').'/oauth/v2/token', [
```
Comment (lucasmichot) [inline src/Neto/Provider.php:49]:
```suggestion
        $response = $this->getHttpClient()->get($this->getTokenUrl(), [
```

File: src/Neto/Provider.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            'headers' => [
+                'X_ACCESS_KEY' => $this->getConfig('client_id'),
+                'X_SECRET_KEY' => $token,
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
```
Comment (lucasmichot) [inline src/Neto/Provider.php:56]:
```suggestion
        return json_decode($response->getBody()->getContents(), true);
```

File: src/Neto/Provider.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
```
Comment (lucasmichot) [inline src/Neto/Provider.php:56]:
```suggestion
        return json_decode((string) $response->getBody(), true);
```

**Review by atymic — APPROVED** [review]

**Pull request merged**